lib/at_host/at_host.c

```diff
@@ -194,6 +195,7 @@ static void uart_rx_handler(uint8_t character)
 	/* Send the command, if there is one to send */
 	if (at_buf[0]) {
 		uart_irq_rx_disable(uart_dev); /* Stop UART to protect at_buf */
+		at_buf_busy = true;
```

Comment:
What if this variable is never set due to the above checks returning prematurely. Could we risk getting stuck in ```isr()```?

Reply:
As long as UART stops providing characters, `isr()` will return properly. `at_buf_busy` shoud only be set when the isr should stop touching `at_buf`.